x-pack/plugins/gis/server/maps_telemetry/maps_telemetry.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+
+function getSavedObjectsClient(server, callCluster) {
+  const { SavedObjectsClient, getSavedObjectsRepository } = server.savedObjects;
+  const internalRepository = getSavedObjectsRepository(callCluster);
+  return new SavedObjectsClient(internalRepository);
+}
+
+function getUniqueLayerCounts(layerCountsList, mapsCount) {
+  const uniqueLayerTypes = _.uniq(_.flatten(
+    layerCountsList.map(lTypes => Object.keys(lTypes))));
+
+  return uniqueLayerTypes.reduce((accu, type) => {
+    const typeCounts = layerCountsList.reduce((accu, tCounts) => {
+      tCounts[type] && accu.push(tCounts[type]);
+      return accu;
+    }, []);
+    accu[type] = {
+      min: _.min(typeCounts),
+      max: _.max(typeCounts),
+      avg: _.sum(typeCounts) / mapsCount
+    };
+    return accu;
+  }, {});
+}
+
+async function buildMapsTelemetry(savedObjectsClient) {
+  const gisMapsSavedObject = await savedObjectsClient.find({
+    type: 'gis-map'
+  });
+
+  const savedMaps = _.get(gisMapsSavedObject, 'saved_objects')
```

Comment:
rename to `layerLists`

Reply:
Funny, that's exactly what it was before but I changed it to savedMaps thinking that was more differentiating and clear here. I'll change it back though